src/storage/types.rs

```diff
@@ -0,0 +1,138 @@
+    pub id: Option<T::DataObjectTypeID>,
+    pub description: Vec<u8>,
+    pub active: bool,
+
+    // TODO in future releases
+    // - replication factor
+    // - storage tranches (empty is ok)
+}
+
+decl_storage! {
+    trait Store for Module<T: Trait> as DataObjectTypeRegistry
+    {
+        // Start at this value
+        pub FirstDataObjectTypeID get(first_data_object_type_id) config(first_data_object_type_id): T::DataObjectTypeID = T::DataObjectTypeID::sa(DEFAULT_FIRST_DATA_OBJECT_TYPE_ID);
+
+        // Increment
+        pub NextDataObjectTypeID get(next_data_object_type_id) build(|config: &GenesisConfig<T>| config.first_data_object_type_id): T::DataObjectTypeID = T::DataObjectTypeID::sa(DEFAULT_FIRST_DATA_OBJECT_TYPE_ID);
+
+        // Mapping of Data object types
+        pub DataObjectTypeMap get(data_object_type): map T::DataObjectTypeID => Option<DataObjectType<T>>;
+    }
+}
+
+decl_event! {
+    pub enum Event<T> where
+        <T as Trait>::DataObjectTypeID
+    {
+        DataObjectTypeAdded(DataObjectTypeID),
+        DataObjectTypeUpdated(DataObjectTypeID),
+    }
+}
+
+
+
+impl<T: Trait> traits::IsActiveDataObjectType<T> for Module<T>
+{
+    fn is_active_data_object_type(which: &T::DataObjectTypeID) -> bool
+    {
+        match Self::ensure_data_object_type(*which)
+        {
+            Ok(dot) => dot.active,
+            Err(_err) => false
+        }
+    }
+}
+
+
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin
+    {
+        fn deposit_event<T>() = default;
+
+        pub fn register_data_object_type(origin, data_object_type: DataObjectType<T>)
+        {
+            ensure_root(origin)?;
+            ensure!(data_object_type.id.is_none(), MSG_REQUIRE_NEW_DOT);
+
+            let new_dot_id = Self::next_data_object_type_id();
+            let dot: DataObjectType<T> = DataObjectType {
+                id: Some(new_dot_id),
+                description: data_object_type.description.clone(),
+                active: data_object_type.active,
+            };
+
+            <DataObjectTypeMap<T>>::insert(new_dot_id, dot);
+            <NextDataObjectTypeID<T>>::mutate(|n| { *n += T::DataObjectTypeID::sa(1); });
+
+            Self::deposit_event(RawEvent::DataObjectTypeAdded(new_dot_id));
+        }
+
+        pub fn update_data_object_type(origin, data_object_type: DataObjectType<T>)
+        {
+            ensure_root(origin)?;
+            ensure!(data_object_type.id.is_some(), MSG_REQUIRE_DOT_ID);
+
+            let id = data_object_type.id.unwrap();
+            let mut dot = Self::ensure_data_object_type(id)?;
+
+            dot.description = data_object_type.description.clone();
+            dot.active = data_object_type.active;
+
+            <DataObjectTypeMap<T>>::insert(id, dot);
+
+            Self::deposit_event(RawEvent::DataObjectTypeUpdated(id));
+        }
+
+        pub fn activate_data_object_type(origin, id: T::DataObjectTypeID, active: bool)
```

Comment:
The function name implies parameter `active` shouldn't be passed. Perhaps remove parameter and add counterpart `fn deactive_data_object_type(origin, id: T::DataObjectTypeID)` each function setting the `active` field correspondigly.

...or changing name to `change_active_status_of_data_object_type(..)`

Reply:
Yeah... I'm not a big fan of multiple functions for the same purpose, nor am I a fan of ultra long function names. I don't much like either suggestion, but I also agree that it could be better.

The correct name IMHO would include "toggle", but to be grammatically correct-ish, the name would then have to be something like `toggle_data_object_type_active_state` and that's super long again.

I think I'll go with a second function. My main objection to that is that if you do that for several flags, you have hundreds of single purpose functions. You can use the `update_...` function for the same thing, though, so this should only be a shortcut for toggling the active state.

Reply:
Also updated.